polyfill/test262-harness.mjs

```diff
@@ -0,0 +1,246 @@
+let passed = 0;
+let skipped = 0;
+
+// === The test loop ===
+
+for (const testFile of testFiles) {
+  // Set up the VM context with the polyfill first, as if it were built-in
+  const testContext = {};
+  vm.createContext(testContext);
+  polyfill.runInContext(testContext);
+
+  // To proceed, we will now need to read the frontmatter
+  const testCode = fs.readFileSync(testFile, UTF8);
+
+  const [, frontmatterString] = frontmatterMatcher.exec(testCode);
+  const frontmatter = yaml.load(frontmatterString);
+
+  const { flags = [], includes = [] } = frontmatter;
+
+  // Load whatever helpers the test specifies. As per the test262 execution
+  // instructions, assert.js and sta.js are always executed even if not
+  // specified, unless the raw flag is given.
+  if (!flags.includes('raw')) includes.unshift('assert.js', 'sta.js');
+  includes.forEach((includeName) => {
+    getHelperScript(includeName).runInContext(testContext);
+  });
+
+  // Various forms of the test's path and filename. testRelPath matches what
+  // is given in the expected failures file. testDisplayName is a slightly
+  // abbreviated form that we use in logging during the run to make it more
+  // likely to fit on one line. progressDisplayName is what's displayed beside
+  // the progress bar: testDisplayName with the actual test filename cut off,
+  // since the individual tests go by too fast to read anyway.
+  const testRelPath = path.relative('test262/test', testFile);
+  const testDisplayName = testRelPath
+    .replace('built-ins/Temporal/', '')
+    .replace('intl402/Temporal/', '(intl) ')
+    .replace('staging/Temporal/', '(stg) ')
+    .replace('/prototype/', '/p/');
+  const progressDisplayName = path.dirname(testDisplayName);
+  progress.tick(0, { test: progressDisplayName });
+
+  // Time each test individually in order to report if they take longer than
+  // 100 ms
+  const testStart = process.hrtime.bigint();
+
+  // Run the test and log a message above the progress bar if the result is not
+  // what it's supposed to be. This is so that you don't have to wait until the
+  // end to see if your test failed.
+  try {
+    vm.runInContext(testCode, testContext);
+    passed++;
+
+    if (expectedFailures.has(testRelPath)) {
+      unexpectedPasses.push(testRelPath);
+      progress.interrupt(`UNEXPECTED PASS: ${testDisplayName}`);
+    }
+  } catch (e) {
+    if (expectedFailures.has(testRelPath)) {
+      skipped++;
+    } else {
+      failures.push({ file: testRelPath, error: e });
+      progress.interrupt(`FAIL: ${testDisplayName}`);
+    }
+  }
+
+  const testFinish = process.hrtime.bigint();
+  const testTime = testFinish - testStart;
+  if (testTime > 100_000_000n) {
+    longTests.push({ file: testRelPath, ns: testTime });
+  }
+
+  progress.tick(1, { test: progressDisplayName });
+}
+
+// === Print results ===
+
+const finish = process.hrtime.bigint();
+const elapsed = Number(finish - start) / 1_000_000_000;
+const failed = failures.length;
```

Comment:
Is this binding carrying its own weight? `failed` reads to me like a boolean, but it is in fact a count. I'd drop it and just use `failures.length` directly.